mpath.go

```diff
@@ -21,94 +30,80 @@ var (
 			s := newScanner()
 			s.sx.Mode = sc.ScanIdents | sc.ScanChars | sc.ScanStrings | sc.ScanRawStrings | sc.ScanComments | sc.SkipComments
 			s.sx.IsIdentRune = func(ch rune, i int) bool {
-				// if i == 0 && unicode.IsDigit(ch) {
-				// 	return false
-				// }
-
-				return ch != '\'' &&
-					ch != '"' &&
-					ch != '(' &&
-					ch != ')' &&
-					ch != '[' &&
-					ch != ']' &&
-					ch != '{' &&
-					ch != '}' &&
-					ch != '@' &&
-					ch != '$' &&
-					ch != '&' &&
-					ch != '.' &&
-					ch != ',' &&
-					ch != '=' &&
-					ch != '>' &&
-					ch != '<' &&
-					ch != '|' &&
-					ch != '!' &&
-					ch != ';' &&
-					ch != '/' &&
-					ch != '*' &&
-					// ch != '?' && // taken out to allow for null propagation
-					!unicode.IsSpace(ch) &&
-					unicode.IsPrint(ch)
+				if invalidRunes[ch] || unicode.IsSpace(ch) {
+					return false
+				}
+
+				return unicode.IsPrint(ch)
 			}
 			s.sx.Error = func(es *sc.Scanner, msg string) {
 				//todo: find a way to pipe this out
 			}
 			return s
 		},
 	}
-	stringsReaderPool = sync.Pool{
-		New: func() any {
-			return &strings.Reader{}
-		},
-	}
 )
 
-func ParseString(ss string) (topOp Operation, err error) {
+// Parse takes an io.ReadSeeker and parses it into an operation tree.
+func Parse(r io.ReadSeeker) (topOp Operation, err error) {
```

Comment:
Maybe change the name so that it follows the convention set by `ParseString` and so that if we add more functions like this, they'll follow that convention:
```suggestion
func ParseReadSeeker(r io.ReadSeeker) (topOp Operation, err error) {
```